### PR TITLE
[2nd attempt] Add WebUSB article

### PR DESCRIPTION
**Pull request by beaufortfrancois**
R: @reillyeon @scheib @owencm

TODO:
- [x] Create http://bit.ly/WebUSBOriginTrial once WebUSB is added to OT form 
- [x] Update https://www.chromestatus.com/features/5651917954875392

### CODE REVIEW

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+The Chrome team is excited to be launching experimental features early to
+developers. The [WebUSB API](https://wicg.github.io/webusb/) is currently a
+draft which means that it is far enough along to be real and usable, but there
+is still time to make fixes that developers need. That's why the Chrome Team is
+actively looking for eager developers to give it a try and send
+[feedback on the spec](https://github.com/wicg/webusb/issues) and
+[feedback on the implementation](https://bugs.chromium.org/p/chromium/issues/entry?components=Blink%3EUSB).
+
+The WebUSB API is currently planned to be enabled experimentally on your
+origin in Origin Trials, and available now locally on your machine using an
```
Comment (scheib) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Link to the Origin Trial section from here?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+drivers and SDKs in order for you (the developer) to take advantage of them.
+Sadly this native code has historically prevented these devices from being used
+by the Web. And that's one of the reasons the WebUSB API has been created: to
+provide a way to expose USB device services to the Web. With this API, hardware
+manufacturers will be able to build cross-platform JavaScript SDKs for their
+devices.
+But most importantly this will **make USB safer and easier to use by bringing
+it to the Web**.
+
+Let's see what you could expect with the WebUSB API:
```
Comment (scheib) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: :+1: Love the new intro.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+API in the field, we will also add this feature in Chrome 54 as an
+[origin trial](https://github.com/jpchase/OriginTrials) for Chrome
+OS, Linux, Mac, and Windows. During the origin trial, the API may still change in
+backward-incompatible ways before we freeze it into the web platform.  To use
+this experimental API in Chrome with no flag, you'll need to [request a token
+for your origin](http://bit.ly/WebUSBOriginTrial) and [insert it in your
+application](https://github.com/jpchase/OriginTrials/blob/gh-pages/developer-guide.md).
+
+The trial will end in March 2017. By that point, we expect to have figured
+out any changes necessary to stabilize the feature and move it out from Origin
```
Comment (scheib) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: We should find a way to correctly explain that either we've finalized it and are shipping it -- or identified a need to run a new experiment, or (very quiet voice) learned not to move forward.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+A second iteration on the WebUSB API will look at [Shared Worker](https://developer.mozilla.org/fr/docs/Web/API/SharedWorker)
+and [Service Worker](https://jakearchibald.github.io/isserviceworkerready/resources.html)
+support. Imagine for instance a security key website using the WebUSB API that
+would install a service worker to act as a middle man to authenticate users.
+
+And for your greatest pleasure, the WebUSB API will come later to Android.
+
+## Resources
+
```
Comment (scheib) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Direct developers a bit more for where to communicate to get help (IRC, mailing list etc?), best practice to file bugs, guide for sharing success & demos (hashtags etc? Have a demos list on the web USB project and accept pull requests to it?).

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+`user` is a member of the `plugdev` group.  Then, just reconnect your device.
+
+## What's next
+
+A second iteration on the WebUSB API will look at [Shared Worker](https://developer.mozilla.org/fr/docs/Web/API/SharedWorker)
+and [Service Worker](https://jakearchibald.github.io/isserviceworkerready/resources.html)
+support. Imagine for instance a security key website using the WebUSB API that
+would install a service worker to act as a middle man to authenticate users.
+
+And for your greatest pleasure, the WebUSB API will come later to Android.
```
Comment (reillyeon) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: No need to wait. It's available now (in Chrome 54)!

**Comment by owencm** [comment]
Looking good! Couple bits of feedback:

I think we can remove "The Chrome team is excited to be launching experimental features early to
 developers. " 

And may I suggest we swap : "The WebUSB API is currently planned to be enabled experimentally on your origin in Origin Trials, and available now locally on your machine using an experimental flag. " for "In the very near future we plan for you to be able to enable WebUSB on your origin via [Origin Trials](https://github.com/jpchase/OriginTrials/blob/gh-pages/developer-guide.md). Until then you can enable it on your local computer for development purposes by flipping an experimental flag.  "

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+But most importantly this will **make USB safer and easier to use by bringing
+it to the Web**.
+
+Let's see what you could expect with the WebUSB API:
+
+1. Buy a USB device,
+2. Plug it into your computer,
+3. A notification appears right away, with the right website to go to for this device.
+4. Simply click on it. Website is there and ready to use!
+5. Click to connect and a USB device chooser shows up in Chrome, where you can pick your device.
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: If I understand this, I get two dialogs: first a notification, then a device chooser. Is that correct.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+manufacturers will be able to build cross-platform JavaScript SDKs for their
+devices.
+But most importantly this will **make USB safer and easier to use by bringing
+it to the Web**.
+
+Let's see what you could expect with the WebUSB API:
+
+1. Buy a USB device,
+2. Plug it into your computer,
+3. A notification appears right away, with the right website to go to for this device.
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: So the USB device is providing a URL that the mobile browser accesses?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+provide a way to expose USB device services to the Web. With this API, hardware
+manufacturers will be able to build cross-platform JavaScript SDKs for their
+devices.
+But most importantly this will **make USB safer and easier to use by bringing
+it to the Web**.
+
+Let's see what you could expect with the WebUSB API:
+
+1. Buy a USB device,
+2. Plug it into your computer,
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: End with periods, not commas.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+3. A notification appears right away, with the right website to go to for this device.
+4. Simply click on it. Website is there and ready to use!
+5. Click to connect and a USB device chooser shows up in Chrome, where you can pick your device.
+6. Tada!
+
+What could it have been after you've plugged your USB device?
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Suggest changing to 'What would this procedure be like without the USB API?"

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 10]
```diff
@@ -0,0 +1,370 @@
...
+
+- Read a box, label, or search online for website -- possibly ending up on the wrong website.
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: change 'search online for website -- possibly ending up' to 'search on line and possibly end up'

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 11]
```diff
@@ -0,0 +1,370 @@
...
+- Have to install a native application.
+- Is it supported on my operating system? Make sure you download the "right" thing.
+- Scary OS prompts popup and warn you about installing drivers/applications from the Internet.
+- Malfunctioning code harms the whole computer, the Web is built to [contain malfunctioning websites](https://www.youtube.com/watch?v=29e0CtgXZSI).
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: This is two sentences, not one. Replace and comma with a period and capitalize appropriately.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 12]
```diff
@@ -0,0 +1,370 @@
...
+- Only use the USB device once? On the Web, the website is gone once you closed tab. On a computer the code sticks around.
+
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: General comment about lists and procedures. Best practice is considered to be making sure they follow a parallel grammatical structure. When a bullet or step has multiple sentences it's the first sentence that follows the parallelism. If an item can't be easily rewritten, it may be because it's a continuation of the previous item. If so, consider merging.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 13]
```diff
@@ -0,0 +1,370 @@
...
+This article assumes you have some basic knowledge of how USB works. If not, I
+recommend reading [USB in a NutShell](http://www.beyondlogic.org/usbnutshell).
+For background information about USB, check out the [official USB
+specifications](http://www.usb.org/home).
+
+The Chrome team is excited to be launching experimental features early to
+developers. The [WebUSB API](https://wicg.github.io/webusb/) is currently a
+draft which means that it is far enough along to be real and usable, but there
+is still time to make fixes that developers need. That's why the Chrome Team is
+actively looking for eager developers to give it a try and send
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: 'give' would be a more appropriate verb in this context than 'send'

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 1 above)
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: 'planned to be enabled experimentally on your origin' 

I don't like the phrasing of this. It makes it sound like we've done something for them we haven't. Simply removing 'on your origin' would fix this.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 14]
```diff
@@ -0,0 +1,370 @@
...
+[feedback on the spec](https://github.com/wicg/webusb/issues) and
+[feedback on the implementation](https://bugs.chromium.org/p/chromium/issues/entry?components=Blink%3EUSB).
+
+The WebUSB API is currently planned to be enabled experimentally on your
+origin in Origin Trials, and available now locally on your machine using an
+experimental flag. The implementation is partially complete and currently
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Never say 'currently available'. Always give a specific Chrome version number.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 15]
```diff
@@ -0,0 +1,370 @@
...
+available on Chrome OS, Linux, Mac, and Windows. Go to
+`chrome://flags/#enable-webusb`, enable the highlighted flag, restart Chrome
+and you should be good to go.
+
+<img style="width:723px; max-height:205px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-flag.png" alt="Web USB Flag highlighted in chrome://flags"/>
+
+## Privacy & Security
+
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Double headings is also not considered a best practice. You need some kind of intro for 'Privacy & Security'. One sentences is adequate.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 16]
```diff
@@ -0,0 +1,370 @@
...
+descriptor](https://wicg.github.io/webusb/#webusb-platform-capability-descriptor).
+Later a [Public Device Registry](https://wicg.github.io/webusb/#public-device-registry) 
+will be created so that hardware manufacturers can support WebUSB on existing
+devices.
+
+### Available for Origin Trials
+
+In order to get as much feedback as possible from developers using the WebUSB
+API in the field, we will also add this feature in Chrome 54 as an
+[origin trial](https://github.com/jpchase/OriginTrials) for Chrome
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: You already covered this. Does it really need to be here twice?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 17]
```diff
@@ -0,0 +1,370 @@
...
+OS, Linux, Mac, and Windows. During the origin trial, the API may still change in
+backward-incompatible ways before we freeze it into the web platform.  To use
+this experimental API in Chrome with no flag, you'll need to [request a token
+for your origin](http://bit.ly/WebUSBOriginTrial) and [insert it in your
+application](https://github.com/jpchase/OriginTrials/blob/gh-pages/developer-guide.md).
+
+The trial will end in March 2017. By that point, we expect to have figured
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Is that date set in stone?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 18]
```diff
@@ -0,0 +1,370 @@
...
+
+### HTTPS Only
+
+Because this experimental API is a powerful new feature added to the Web,
+Chrome aims to make it available only to [secure
+contexts](https://w3c.github.io/webappsec/specs/powerfulfeatures/#intro). This
+means you'll need to build with TLS in mind.
+
+> We care deeply about security, so you will notice that new Web capabilities
+require HTTPS. The WebUSB API is no different, and is yet another good reason
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: I thought we had a published policy on Chromium.org somewhere. Consider linking to it.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 19]
```diff
@@ -0,0 +1,370 @@
...
+.then(() => device.selectConfiguration(1)) // Select configuration #1 for the device.
+.then(() => device.claimInterface(2)) // Request exclusive control over interface #2.
+.then(() => device.controlTransferOut({
+    requestType: 'class',
+    recipient: 'interface',
+    request: 0x22,
+    value: 0x01,
+    index: 0x02})) // Ready to receive data
+.then(() => device.transferIn(5, 64)) // Waiting for 64 bytes of data from endpoint #5.
+.then(result => {
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Where would I learn about the members of device?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 20]
```diff
@@ -0,0 +1,370 @@
...
+  { 1, "webusb.github.io/arduino/demos/" },
+  { 0, "localhost:8000" },
+};
+
+const uint8_t ALLOWED_ORIGINS[] = { 1, 2 };
+
+WebUSB WebUSBSerial(URLS, 2, 1, ALLOWED_ORIGINS, 2);
+
+#define Serial WebUSBSerial
+
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: This section needs an an intro. One sentence is adequate.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 21]
```diff
@@ -0,0 +1,370 @@
...
+The third-party [WebUSB Arduino
+library](https://github.com/webusb/arduino/tree/gh-pages/library/WebUSB) used
+in the sample code above does basically two things:
+
+- The device acts as a WebUSB device enabling Chrome to read
+  the [landing page URL](https://wicg.github.io/webusb/#webusb-platform-capability-descriptor)
+  and the [list of origins](https://wicg.github.io/webusb/#get-allowed-origins) allowed to communicate with.
+- It exposes a WebUSB Serial API that you may use to override the default one.
+
+Let's focus back to the JavaScript code. Once we get the `device` picked by the
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: 'Let's focus back to the JavaScript code.' Change to 'Let's look at the JavaScript code again.'

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 19 above)
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: I just found where you describe those. Find some place to mention that you'll cover these calls later in the article.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 22]
```diff
@@ -0,0 +1,370 @@
...
+  let decoder = new TextDecoder();
+  console.log('Received: ' + decoder.decode(result.data));
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+Please keep in mind that the WebUSB library we are using here is just
+implementing one example protocol (based on the standard USB serial protocol)
+and that you can create any set and types of endpoints you
+wish. Control transfers are especially nice for small configuration commands as
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Generally in this article you're talking to users of USB devices. This sentence seems to be talking to manufacturers of USB devices. Don't shift audiences. Instead of 'you can create any set and types of endpoints you wish' say something like 'manufacturers can create any set and types of endpoints they wish'.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 23]
```diff
@@ -0,0 +1,370 @@
...
+with the USB device. Then, we have to select an available USB Configuration
+with `device.selectConfiguration`. Remember that a Configuration specifies how
+the device is powered, its maximum power consumption and its number of
+interfaces. Talking about interfaces, we also need to request exclusive access
+with `device.claimInterface` since data can only be transferred to an interface
+or associated endpoints when the interface is claimed. Finally calling
+`device.controlTransferOut` is needed to set up the Arduino device with the
+appropriate commands to communicate through the WebUSB Serial API.
+
+From there, `device.transferIn` performs a Bulk Transfer onto the
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Change 'Bulk Transfer' to 'bulk transfer'.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 24]
```diff
@@ -0,0 +1,370 @@
...
+device to inform that the host is ready to receive bulk data. Then, the promise
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: To inform who?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 25]
```diff
@@ -0,0 +1,370 @@
...
+For those who are familiar with USB, all of this should look pretty familiar.
+
+### I want moar
+
+The WebUSB API lets you interact with the all USB transfer/endpoint types:
+
+- CONTROL transfers, used to send or receive configuration or command
+  parameters to a USB device are handled with `controlTransferIn(setup,
+  length)` and `controlTransferOut(setup, data)`.
+- INTERRUPT transfers, used for small amount of time sensitive data are handled
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Change 'for small' to 'for a small'.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 26]
```diff
@@ -0,0 +1,370 @@
...
+  with same methods as BULK transfers with `transferIn(endpointNumber, length)`
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Change 'with same' to 'with the same'.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 27]
```diff
@@ -0,0 +1,370 @@
...
+  and `transferOut(endpointNumber, data)`.
+- ISOCHRONOUS transfers, used for streams of data, like video and sound are
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Remove the second comma.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 28]
```diff
@@ -0,0 +1,370 @@
...
+ `transferOut(endpointNumber, data)`.
+
+You may also want to have a look at Mike Tsao's [WebLight
+project](https://github.com/sowbug/weblight) which provides a ground-up example
+of building a USB-controlled LED device designed for the WebUSB API (not using
+an Arduino here). You'll find hardware, software, and firmware.
+
+## Tips
+
+Debugging USB in Chrome is made easier with the internal page
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Remove the word made.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 29]
```diff
@@ -0,0 +1,370 @@
...
+`chrome://device-log` where you can see all USB device related events in one
+single place.
+
+<img style="width:723px; max-height:414px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-device-log-page.png" alt="Internal page to debug Web USB"/>
+
+Early adopters who want to test their existing devices with WebUSB before they
+can update their firmware or the [Public Device Registry](https://wicg.github.io/webusb/#public-device-registry)
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Remove the word can.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 30]
```diff
@@ -0,0 +1,370 @@
...
+Debugging USB in Chrome is made easier with the internal page
+`chrome://device-log` where you can see all USB device related events in one
+single place.
+
+<img style="width:723px; max-height:414px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-device-log-page.png" alt="Internal page to debug Web USB"/>
+
+Early adopters who want to test their existing devices with WebUSB before they
+can update their firmware or the [Public Device Registry](https://wicg.github.io/webusb/#public-device-registry) 
+is [implemented](https://bugs.chromium.org/p/chromium/issues/detail?id=598765) 
+are not out ouf luck there thanks to a dedicated switch. To disable checking of
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Remove the word there.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown [hunk 31]
```diff
@@ -0,0 +1,370 @@
...
+SUBSYSTEM=="usb", ATTR{idVendor}=="[yourdevicevendor]", MODE="0664", GROUP="plugdev"
+```
+
+where `[yourdevicevendor]` is `2341` if your device is an Arduino for instance.
+`ATTR{idProduct}` can also be added for a more specific rule. Make sure your
+`user` is a member of the `plugdev` group.  Then, just reconnect your device.
+
+## What's next
+
+A second iteration on the WebUSB API will look at [Shared Worker](https://developer.mozilla.org/fr/docs/Web/API/SharedWorker)
```
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Change on to of.

**Comment by jpmedley** [comment]
Here's some additional comments.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 6 above)
Comment (reillyeon) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Yes, desktop browsers will display a notification when the device is connected. The site calling navigator.usb.requestDevice() displays a permission prompt.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 7 above)
Comment (reillyeon) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: The USB device provides a landing page URL and a set of allowed origins. Unfortunately the Android OS's permission model prevents us from reading the landing page URL from the device early enough to display it.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 22 above)
Comment (reillyeon) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: The article should be addressed to manufacturers.

Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: I shouldn't have used the word users when I meant web developers who build pages to take advantage of USB devices. I hadn't noticed any indication that manufacturers were intended up to this point. It's appearance in this paragraph feels out of place.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 7 above)
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Let me rephrase the question. The notification contains a URL that the user can open?

Comment (reillyeon) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Yes.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 1 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Both done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 3 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: How would you express that?
@owencm WDYT?

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 5 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Woohoo!
Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 7 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: See https://github.com/google/WebFundamentals/blob/c6c4e80c008c99b7ea8aedcc86a5c56e0ec28f3c/src/content/en/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-notification.png

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 8 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 9 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 10 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 11 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 13 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 14 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: This makes it harder for a random user reading this to know if the API is available or not as he/she has to find out which version he/she's actually running.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 15 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: It still looks good (see below). Which sentence would you add then?

screenshot 2016-08-17 at 11 54 00 am

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 16 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Yes. We want to link to this specific advanced section when people ask about Origin Trials and we also want to call it out at the very beginning.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 17 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: It will yes like other OT APIs. See https://github.com/jpchase/OriginTrials/blob/gh-pages/available-trials.md

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 18 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: I've been asked to point to this page instead of http://www.chromium.org/Home/chromium-security/deprecating-powerful-features-on-insecure-origins previously.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 20 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: This is not a section. `#define` is a C component that allows the programmer to give a name to a constant value before the program is compiled.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 19 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 21 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 22 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: You're right @jpmedley, I should not have shifted audiences hence why I've now replaced `you` with `manufacturers`. As noted by @reillyeon, this article is for both: web developers and manufacturers, so I'd like to keep it there.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 23 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 24 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: The device.
Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 25 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 26 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 27 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 28 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 29 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 30 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 31 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 4 above)
Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: Done.

Not sure about best practice to file bugs though. 
@reillyeon may help there.

**Comment by beaufortfrancois** [comment]
@jpmedley @reillyeon @scheib @owencm I've addressed your feedback and left some questions.
Thank you for reviewing this!

**Comment by beaufortfrancois** [comment]
Live preview at https://pr-3218-dot-weasel-dot-web-central.appspot.com/web/updates/2016/03/access-usb-devices-on-the-web?hl=en

File: src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown (hunk 15 above)
Comment (jpmedley) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: It's not obvious how origin trials fit with USB devices. Either explain the connection or remove 'Privacy & Security' and promote the next two headings to level 2.

Comment (beaufortfrancois) [inline src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown]: You're right.
Fixed in https://github.com/google/WebFundamentals/pull/3218/commits/58cf8753fd3c5cdcd9e1dc88166e0e9f82dc8ac9

**Comment by reillyeon** [comment]
lgtm

**Comment by beaufortfrancois** [comment]
@petele Can you merge and push this live?
